🌱 Prepare for v2 beta1 release

azeemshaikh38 opened the pull request:
Commit created from this PR will be used to generate the pre-release of scorecard-action. I've decided to name the tag - `v2.0.0-beta.1`.

justaugustus commented on action.yaml:
```diff
@@ -48,6 +48,6 @@ branding:
 runs:
   using: "docker"
-  image: "docker://gcr.io/openssf/scorecard-action:latest"
+  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta1"
```
```suggestion
  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta.1"
```

Needs a `.` after `beta` to be a SemVer-compliant tag.

justaugustus commented:
(Let's also see if we can drain the remaining dependabot updates before cutting the release)

azeemshaikh38 commented:
> (Let's also see if we can drain the remaining dependabot updates before cutting the release)

Fair point. Looking into it.

laurentsimon commented on RELEASE.md:
```diff
@@ -30,6 +30,32 @@ We're not striving for perfection with the template, but the tracking issue
+
+Pre-select the tag which will be used for the release. For this document, we'll use: `Tag`.
+
+Update the image tage in [action.yaml](action.yaml) to use `Tag`.
```
nit: tage

azeemshaikh38 replied:
Done, thanks.

laurentsimon commented on action.yaml:
```diff
@@ -48,6 +48,6 @@ branding:
 runs:
   using: "docker"
-  image: "docker://gcr.io/openssf/scorecard-action:latest"
+  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta.1"
```
this will be replaced by a hash, like `:tag@sha256:xxx` after?

naveensrinivasan commented:
> > > (Let's also see if we can drain the remaining dependabot updates before cutting the release)
> > 
> > 
> > Fair point. Looking into it.
> 
> Getting errors running `go mod tidy`:
> 
> ```
> go mod tidy
> go: finding module for package go.opentelemetry.io/otel/metric/registry
> go: finding module for package go.opentelemetry.io/otel/semconv
> github.com/ossf/scorecard-action/signing imports
>         github.com/sigstore/cosign/cmd/cosign/cli/sign imports
>         github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioverifier/ctl imports
>         github.com/google/certificate-transparency-go imports
>         go.etcd.io/etcd/v3 imports
>         go.etcd.io/etcd/tests/v3/integration imports
>         go.etcd.io/etcd/server/v3/embed imports
>         go.opentelemetry.io/otel/semconv: module go.opentelemetry.io/otel@latest found (v1.8.0), but does not contain package go.opentelemetry.io/otel/semconv
> github.com/ossf/scorecard-action/signing imports
>         github.com/sigstore/cosign/cmd/cosign/cli/sign imports
>         github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioverifier/ctl imports
>         github.com/google/certificate-transparency-go imports
>         go.etcd.io/etcd/v3 imports
>         go.etcd.io/etcd/tests/v3/integration imports
>         go.etcd.io/etcd/server/v3/embed imports
>         go.opentelemetry.io/otel/exporters/otlp imports
>         go.opentelemetry.io/otel/sdk/metric/controller/basic imports
>         go.opentelemetry.io/otel/metric/registry: module go.opentelemetry.io/otel/metric@latest found (v0.31.0), but does not contain package go.opentelemetry.io/otel/metric/registry
> ```
> 
> Anyone have an idea?

Is this on your branch?

azeemshaikh38 commented:
> Is this on your branch?

No, this is on HEAD.

naveensrinivasan commented:
> > Is this on your branch?
> 
> No, this is on HEAD.

Looks like this is the problem https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999

azeemshaikh38 commented:
> > > Is this on your branch?
> > 
> > 
> > No, this is on HEAD.
> 
> Looks like this is the problem [open-telemetry/opentelemetry-go#2938 (comment)](https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999)

Thanks, that's a great find! I'll try to see if I can use the comments there to resolve our dependency problem.

naveensrinivasan commented:
> > > > Is this on your branch?
> > > 
> > > 
> > > No, this is on HEAD.
> > 
> > 
> > Looks like this is the problem [open-telemetry/opentelemetry-go#2938 (comment)](https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999)
> 
> Thanks, that's a great find! I'll try to see if I can use the comments there to resolve our dependency problem.

It is a `replace` directive as the project has broken the API, and the downstream hasn't changed. This is dependency hell.

naveensrinivasan commented:
> > (Let's also see if we can drain the remaining dependabot updates before cutting the release)
> 
> Fair point. Looking into it.

Most of them have been merged!

justaugustus commented:
> @justaugustus blocked on your approval. Dependabot alerts are now fixed.

Azeem, can you investigate the unit test failures?
I believe it's related to the signing logic.

azeemshaikh38 commented:
> > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> 
> Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.

They are intermittent failures since the issue is with the REST API the test talks to (its a mock request, so the failures are only happening on the test but prod works fine). The fix is going to be on the API side, no changes needed here. Will retry the test to get it to pass.

azeemshaikh38 commented:
> > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> 
> Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.

Fixed now with the retry :)

azeemshaikh38 commented:
> > > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> > 
> > 
> > Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.
> 
> Fixed now with the retry :)

@justaugustus friendly ping.

justaugustus commented:
> @justaugustus friendly ping.

Thanks for the details on the failure.
Approved!
